src/player.cpp

```diff
@@ -4229,6 +4232,28 @@ bool player::has_pda()
     return pda;
 }
 
+
+bool player::has_alarm_clock()
+{
+    return ( has_item_with_flag("ALARMCLOCK") ||
+             ( 
+               ( g->m.veh_at( posx(), posy() ) != nullptr ) && 
+               g->m.veh_at( posx(), posy() )->all_parts_with_feature( "ALARMCLOCK", true ).size()>0
```

Comment:
noted. copied that from elsewhere in the code. didn't know about std::vector.empty